ads/unruly.js

```diff
@@ -0,0 +1,28 @@
+/*
+  Copyright 2015 The AMP HTML Authors. All Rights Reserved.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS-IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+
+import {loadScript} from '../3p/3p';
+
+export function unruly(global, data, scriptLoader = loadScript) {
+  global.unruly = global.unruly || {};
+  global.unruly.native = {
```

Comment:
You may want to validate the input here. We have this function to help:
https://github.com/ampproject/amphtml/blob/d86f11f72e40ab6f242b3e5d578d6bac26f5b192/3p/3p.js#L213-L227